source/IntegrationTests/Infrastructure.CimMessageAdapter/Messages/RequestAggregatedMeasureData/RequestAggregatedMeasureDataReceiverTests.cs

```diff
@@ -498,30 +503,42 @@ public async Task Business_type_is_not_allowed()
             .Message();
 
         var messageParserResult = await ParseMessageAsync(message);
-        var result = await _requestAggregatedMeasureDataValidator.ValidateAsync(messageParserResult.Dto!, CancellationToken.None);
+        var result = await _requestAggregatedMeasureDataMessageValidator.ValidateAsync((RequestAggregatedMeasureDataMessage)messageParserResult.IncomingMessage!, CancellationToken.None);
 
         Assert.Contains(result.Errors, error => error is NotSupportedBusinessType);
     }
 
-    private RequestAggregatedMeasureDataDto CreateMarketMessageWithAuthentication(RequestAggregatedMeasureDataDto dto, string knownSenderId, string knownSenderRole)
+    private RequestAggregatedMeasureDataDto CreateMarketMessageWithAuthentication(RequestAggregatedMeasureDataMessage message, string knownSenderId, string knownSenderRole)
     {
         var authenticatedActor = GetService<AuthenticatedActor>();
         authenticatedActor.SetAuthenticatedActor(new ActorIdentity(ActorNumber.Create(knownSenderId), restriction: Restriction.None,  ActorRole.FromCode(knownSenderRole)));
 
+        var series = message.Series
+            .Select(
+                serie => new Serie(
+                    serie.Id,
+                    serie.MarketEvaluationPointType,
+                    serie.MarketEvaluationSettlementMethod,
+                    serie.StartDateAndOrTimeDateTime,
+                    serie.EndDateAndOrTimeDateTime,
+                    serie.MeteringGridAreaDomainId,
+                    serie.EnergySupplierMarketParticipantId,
+                    serie.BalanceResponsiblePartyMarketParticipantId,
+                    serie.SettlementSeriesVersion)).ToList().AsReadOnly();
         return new RequestAggregatedMeasureDataDto(
-            dto.SenderNumber,
-            dto.SenderRoleCode,
-            dto.ReceiverNumber,
-            dto.ReceiverRoleCode,
-            dto.BusinessReason,
-            dto.MessageType,
-            dto.MessageId,
-            dto.CreatedAt,
-            dto.BusinessType,
-            dto.Series);
+            message.SenderNumber,
+            message.SenderRoleCode,
+            message.ReceiverNumber,
+            message.ReceiverRoleCode,
+            message.BusinessReason,
+            message.MessageType,
+            message.MessageId,
+            message.CreatedAt,
+            message.BusinessType,
+            series);
     }
 
-    private Task<RequestAggregatedMeasureDataMarketMessageParserResult> ParseMessageAsync(Stream message)
+    private Task<IncomingMarketMessageParserResult> ParseMessageAsync(Stream message)
```

Comment:
Could be something like below, to avoid having to cast to `RequestAggregatedMeasureDataMessage` in every test
```suggestion
    private Task<(RequestAggregatedMeasureDataMessage? ParsedMessage, IncomingMarketMessageParserResult ParseResult)> ParseMessageAsync(Stream message)
```